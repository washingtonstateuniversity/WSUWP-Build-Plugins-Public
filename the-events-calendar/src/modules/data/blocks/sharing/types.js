/**
 * Internal dependencies
 */
import { PREFIX_EVENTS_STORE } from '@moderntribe/events/data/utils';

export const SET_INITIAL_STATE = `${ PREFIX_EVENTS_STORE }/SHARING_BLOCK/SET_INITIAL_STATE`;

export const SET_GOOGLE_CALENDAR_LABEL = `${ PREFIX_EVENTS_STORE }/SET_GOOGLE_CALENDAR_LABEL`;

export const SET_ICAL_LABEL = `${ PREFIX_EVENTS_STORE }/SET_ICAL_LABEL`;

export const SET_HAS_ICAL = `${ PREFIX_EVENTS_STORE }/SET_HAS_ICAL`;
export const TOGGLE_ICAL = `${ PREFIX_EVENTS_STORE }/TOGGLE_ICAL`;

export const SET_HAS_GOOGLE_CALENDAR = `${ PREFIX_EVENTS_STORE }/SET_HAS_GOOGLE_CALENDAR`;
export const TOGGLE_GOOGLE_CALENDAR = `${ PREFIX_EVENTS_STORE }/TOGGLE_GOOGLE_CALENDAR`;
