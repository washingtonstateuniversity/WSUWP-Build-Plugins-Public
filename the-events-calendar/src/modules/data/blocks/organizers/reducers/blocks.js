/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import * as types from './../types';
import block from './block';

export const byId = ( state = {}, action ) => {
	switch ( action.type ) {
		case types.ADD_ORGANIZER_BLOCK:
		case types.REMOVE_ORGANIZER_BLOCK:
			return {
				...state,
				[ action.payload.id ]: block( state[ action.payload.id ], action ),
			};
		default:
			return state;
	}
};

export const allIds = ( state = [], action ) => {
	switch ( action.type ) {
		case types.ADD_ORGANIZER_BLOCK:
			return [ ...state, action.payload.organizer ];
		case types.REMOVE_ORGANIZER_BLOCK:
			return [ ...state ].filter( ( organizer ) => organizer !== action.payload.organizer );
		default:
			return state;
	}
};

export default combineReducers( {
	byId,
	allIds,
} );
