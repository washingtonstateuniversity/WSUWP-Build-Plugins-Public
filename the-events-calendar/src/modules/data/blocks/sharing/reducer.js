/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';

/**
 * Internal dependencies
 */
import * as types from './types';

export const DEFAULT_STATE = {
	googleCalendarLabel: __( 'Google Calendar', 'the-events-calendar' ),
	iCalLabel: __( 'iCal Export', 'the-events-calendar' ),
	hasiCal: true,
	hasGoogleCalendar: true,
};

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case types.SET_GOOGLE_CALENDAR_LABEL:
			return {
				...state,
				googleCalendarLabel: action.payload.label,
			};
		case types.SET_ICAL_LABEL:
			return {
				...state,
				iCalLabel: action.payload.label,
			};
		case types.SET_HAS_GOOGLE_CALENDAR:
			return {
				...state,
				hasGoogleCalendar: action.payload.hasGoogleCalendar,
			};
		case types.TOGGLE_GOOGLE_CALENDAR:
			return {
				...state,
				hasGoogleCalendar: ! state.hasGoogleCalendar,
			};
		case types.SET_HAS_ICAL:
			return {
				...state,
				hasiCal: action.payload.hasiCal,
			};
		case types.TOGGLE_ICAL:
			return {
				...state,
				hasiCal: ! state.hasiCal,
			};
		default:
			return state;
	}
};
