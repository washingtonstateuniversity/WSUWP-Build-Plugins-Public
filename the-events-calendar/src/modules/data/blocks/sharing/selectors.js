/**
 * External dependencies
 */
import { createSelector } from 'reselect';

export const sharingSelector = ( state ) => state.events.blocks.sharing;

export const googleCalendarLabelSelector = createSelector(
	[ sharingSelector ],
	( values ) => values.googleCalendarLabel,
);

export const hasGooglecalendarLabel = createSelector(
	[ sharingSelector ],
	( values ) => values.hasGoogleCalendar,
);

export const iCalLabelSelector = createSelector(
	[ sharingSelector ],
	( values ) => values.iCalLabel,
);

export const hasIcalSelector = createSelector(
	[ sharingSelector ],
	( values ) => values.hasiCal,
);
