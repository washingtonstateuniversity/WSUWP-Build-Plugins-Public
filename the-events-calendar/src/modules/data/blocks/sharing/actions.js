/**
 * Internal dependencies
 */
import * as types from './types';
import { DEFAULT_STATE } from './reducer';

export const setGoogleCalendarLabel = ( label ) => ( {
	type: types.SET_GOOGLE_CALENDAR_LABEL,
	payload: {
		label,
	},
} );

export const setHasGoogleCalendar = ( hasGoogleCalendar ) => ( {
	type: types.SET_HAS_GOOGLE_CALENDAR,
	payload: {
		hasGoogleCalendar,
	},
} );

export const toggleGoogleCalendar = () => ( { type: types.TOGGLE_GOOGLE_CALENDAR } );

export const setiCalLabel = ( label ) => ( {
	type: types.SET_ICAL_LABEL,
	payload: {
		label,
	},
} );

export const setHasIcal = ( hasiCal ) => ( {
	type: types.SET_HAS_ICAL,
	payload: {
		hasiCal,
	},
} );

export const toggleIcalLabel = () => ( { type: types.TOGGLE_ICAL } );

export const setInitialState = ( props ) => ( {
	type: types.SET_INITIAL_STATE,
	payload: props,
} );
