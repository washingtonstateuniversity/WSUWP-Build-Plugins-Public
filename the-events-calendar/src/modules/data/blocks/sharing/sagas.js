/**
 * External Dependencies
 */
import { put, all, takeEvery } from 'redux-saga/effects';

/**
 * Internal dependencies
 */
import * as types from './types';
import { DEFAULT_STATE } from './reducer';
import * as actions from './actions';

export function* setInitialState( action ) {
	const { get } = action.payload;

	yield all( [
		put(
			actions.setGoogleCalendarLabel(
				get( 'googleCalendarLabel', DEFAULT_STATE.googleCalendarLabel )
			)
		),
		put( actions.setiCalLabel( get( 'iCalLabel', DEFAULT_STATE.iCalLabel ) ) ),
		put( actions.setHasIcal( get( 'hasiCal', DEFAULT_STATE.hasiCal ) ) ),
		put(
			actions.setHasGoogleCalendar( get( 'hasGoogleCalendar', DEFAULT_STATE.hasGoogleCalendar ) )
		),
	] );
}

export default function* watchers() {
	yield takeEvery( types.SET_INITIAL_STATE, setInitialState );
}
