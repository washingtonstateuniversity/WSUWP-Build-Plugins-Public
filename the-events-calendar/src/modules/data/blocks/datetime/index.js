/**
 * Internal dependencies
 */
import reducer from './reducer';
import * as types from './types';
import * as actions from './actions';
import * as thunks from './thunks';
import * as selectors from './selectors';
import sagas from './sagas';

export default reducer;
export { types, actions, thunks, selectors, sagas };
