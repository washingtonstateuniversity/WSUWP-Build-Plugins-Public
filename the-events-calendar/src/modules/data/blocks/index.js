/**
 * Internal dependencies
 */
import * as datetime from './datetime';

export { datetime };
export { default } from './reducer';
