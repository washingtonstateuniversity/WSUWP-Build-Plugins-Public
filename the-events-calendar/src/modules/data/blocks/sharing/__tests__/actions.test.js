/**
 * External dependencies
 */
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
/**
 * Internal dependencies
 */
import { actions } from '@moderntribe/events/data/blocks/sharing';

const middlewares = [ thunk ];
const mockStore = configureStore( middlewares );

describe( '[STORE] - Sharing actions', () => {
	it( 'Should set initial state', () => {
		expect( actions.setInitialState( {} ) ).toMatchSnapshot();
	} );

	it( 'Should set the iCal Label', () => {
		expect( actions.setiCalLabel( 'Modern Tribe iCal' ) ).toMatchSnapshot();
	} );

	it( 'Should set the Google calendar label', () => {
		expect( actions.setGoogleCalendarLabel( 'Modern Tribe Google Calendar' ) ).toMatchSnapshot();
	} );

	it( 'Should set if has the google calendar', () => {
		expect( actions.setHasGoogleCalendar( true ) ).toMatchSnapshot();
		expect( actions.setHasGoogleCalendar( false ) ).toMatchSnapshot();
	} );

	it( 'Should set if has the iCal Calendar', () => {
		expect( actions.setHasIcal( true ) ).toMatchSnapshot();
		expect( actions.setHasIcal( false ) ).toMatchSnapshot();
	} );

	it( 'Should toggle if has the google label', () => {
		expect( actions.toggleGoogleCalendar() ).toMatchSnapshot();
	} );

	it( 'Should toggle if has the iCal label', () => {
		expect( actions.toggleIcalLabel() ).toMatchSnapshot();
	} );
} );
