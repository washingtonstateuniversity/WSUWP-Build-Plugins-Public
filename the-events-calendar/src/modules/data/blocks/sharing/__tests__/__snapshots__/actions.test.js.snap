// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[STORE] - Sharing actions Should set if has the google calendar 1`] = `
Object {
  "payload": Object {
    "hasGoogleCalendar": true,
  },
  "type": "@@MT/EVENTS/SET_HAS_GOOGLE_CALENDAR",
}
`;

exports[`[STORE] - Sharing actions Should set if has the google calendar 2`] = `
Object {
  "payload": Object {
    "hasGoogleCalendar": false,
  },
  "type": "@@MT/EVENTS/SET_HAS_GOOGLE_CALENDAR",
}
`;

exports[`[STORE] - Sharing actions Should set if has the iCal Calendar 1`] = `
Object {
  "payload": Object {
    "hasiCal": true,
  },
  "type": "@@MT/EVENTS/SET_HAS_ICAL",
}
`;

exports[`[STORE] - Sharing actions Should set if has the iCal Calendar 2`] = `
Object {
  "payload": Object {
    "hasiCal": false,
  },
  "type": "@@MT/EVENTS/SET_HAS_ICAL",
}
`;

exports[`[STORE] - Sharing actions Should set initial state 1`] = `
Object {
  "payload": Object {},
  "type": "@@MT/EVENTS/SHARING_BLOCK/SET_INITIAL_STATE",
}
`;

exports[`[STORE] - Sharing actions Should set the Google calendar label 1`] = `
Object {
  "payload": Object {
    "label": "Modern Tribe Google Calendar",
  },
  "type": "@@MT/EVENTS/SET_GOOGLE_CALENDAR_LABEL",
}
`;

exports[`[STORE] - Sharing actions Should set the iCal Label 1`] = `
Object {
  "payload": Object {
    "label": "Modern Tribe iCal",
  },
  "type": "@@MT/EVENTS/SET_ICAL_LABEL",
}
`;

exports[`[STORE] - Sharing actions Should toggle if has the google label 1`] = `
Object {
  "type": "@@MT/EVENTS/TOGGLE_GOOGLE_CALENDAR",
}
`;

exports[`[STORE] - Sharing actions Should toggle if has the iCal label 1`] = `
Object {
  "type": "@@MT/EVENTS/TOGGLE_ICAL",
}
`;
